source/s3_client.c

```diff
@@ -1028,63 +1026,75 @@ static struct aws_s3_meta_request *s_s3_client_meta_request_factory_default(
+                            " bytes is too large for platform.",
+                            part_size_uint64);
 
-                size_t part_size = (size_t)part_size_uint64;
+                        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+                        return NULL;
+                    }
 
-                if (part_size > client_max_part_size) {
-                    AWS_LOGF_ERROR(
-                        AWS_LS_S3_META_REQUEST,
-                        "Could not create auto-ranged-put meta request; required part size for put request is %" PRIu64
-                        ", but current maximum part size is %" PRIu64,
-                        (uint64_t)part_size,
-                        (uint64_t)client_max_part_size);
-                    aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
-                    return NULL;
-                }
+                    part_size = (size_t)part_size_uint64;
 
-                if (part_size < client_part_size) {
-                    part_size = client_part_size;
-                }
-                if (content_length < part_size) {
-                    /* When the content length is smaller than part size and larger than the threshold, we set one part
-                     * with the whole length */
-                    part_size = (size_t)content_length;
-                }
+                    if (part_size > client_max_part_size) {
+                        AWS_LOGF_ERROR(
+                            AWS_LS_S3_META_REQUEST,
+                            "Could not create auto-ranged-put meta request; required part size for put request is "
+                            "%" PRIu64 ", but current maximum part size is %" PRIu64,
+                            (uint64_t)part_size,
+                            (uint64_t)client_max_part_size);
+                        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+                        return NULL;
+                    }
+
+                    if (part_size < client_part_size) {
+                        part_size = client_part_size;
+                    }
+                    if (content_length < part_size) {
+                        /* When the content length is smaller than part size and larger than the threshold, we set one
+                         * part with the whole length */
+                        part_size = (size_t)content_length;
+                    }
 
-                uint32_t num_parts = (uint32_t)(content_length / part_size);
+                    num_parts = (uint32_t)(content_length / part_size);
 
-                if ((content_length % part_size) > 0) {
-                    ++num_parts;
+                    if ((content_length % part_size) > 0) {
+                        ++num_parts;
+                    }
                 }
 
                 return aws_s3_meta_request_auto_ranged_put_new(
                     client->allocator, client, part_size, content_length, num_parts, options);
             } else {
```

Comment:
trivial: the if statement was like 100 lines ago
```suggestion
            } else { /* else using resume token */
```